pkg/fqdn/dnsproxy/helpers.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnsproxy
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/maps/proxymap"
+	"github.com/cilium/cilium/pkg/u8proto"
+	"github.com/miekg/dns"
+)
+
+// simpleFQDNCheck matches plain DNS names
+// See https://en.wikipedia.org/wiki/Hostname
+var simpleFQDNCheck = regexp.MustCompile("^[-a-zA-Z0-9.]*[.]$")
+
+// escapeSimpleFQDN escapes the dots in simple DNS names to avoid spurious
+// matches.
+func escapeSimpleFQDN(re string) string {
```

Comment:
I'm fine making these regexes! That doesn't remove the need for the equivalent code elsewhere:
- DNSPoller is still in the code, and it might stay there (I don't particularly care either way, but I can discuss the situations where it would be useful to keep). If it does stay, it can only poll simple FQDNs (as defined above this). This isn't what you are objecting to, however.
- The main reason to do the magic is that the current toFQDNs takes literal FQDNs in `matchName`. This PR makes those regexy and that might be risky if a user types in `matchName: foo.com` expecting the '.' to be a '.' and not any-character. One way around this is to add a `matchRegex` instead.
As a final note, we also add a trailing '.' in many cases, since foo.com as a literal is also incorrect (but no one remembers to add them). This makes sense in the literal case because a trailing dot always occurs in the actual DNS lookup.